src/query/interpreter.cpp

```diff
@@ -1546,73 +1546,176 @@ std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreat
+          ++it;
+        }
       }
     }
-  }
-  // Iterate over all indexed vertices
-  std::for_each(indices_info.begin(), indices_info.end(), [execution_db_accessor, &counter](const LPIndex &index_info) {
-    auto vertices = execution_db_accessor->Vertices(storage::View::OLD, index_info.first, index_info.second);
-    std::for_each(vertices.begin(), vertices.end(), [&index_info, &counter](const auto &vertex) {
-      counter[index_info][*vertex.GetProperty(storage::View::OLD, index_info.second)]++;
-    });
-  });
+  };
 
-  results.reserve(counter.size());
-  std::for_each(counter.begin(), counter.end(), [&results, execution_db_accessor](const auto &counter_entry) {
-    const auto &[label_property, values_map] = counter_entry;
-    std::vector<TypedValue> result;
-    result.reserve(kDeleteStatisticsNumResults);
-    // Extract info
-    int64_t count_property_value = std::accumulate(
-        values_map.begin(), values_map.end(), 0,
-        [](int64_t prev_value, const auto &prop_value_count) { return prev_value + prop_value_count.second; });
-    // num_distinc_values will never be 0
-    double avg_group_size = static_cast<double>(count_property_value) / static_cast<double>(values_map.size());
-    double chi_squared_stat = std::accumulate(
-        values_map.begin(), values_map.end(), 0.0, [avg_group_size](double prev_result, const auto &value_entry) {
-          return prev_result + utils::ChiSquaredValue(value_entry.second, avg_group_size);
+  auto erase_not_specified_label_property_indices = [&labels, execution_db_accessor](auto &index_info) {
+    if (labels[0] != kAsterisk) {
+      for (auto it = index_info.cbegin(); it != index_info.cend();) {
+        if (std::find(labels.begin(), labels.end(), execution_db_accessor->LabelToName(it->first)) == labels.end()) {
+          it = index_info.erase(it);
+        } else {
+          ++it;
+        }
+      }
+    }
+  };
+
+  auto populate_label_stats = [execution_db_accessor, view](auto index_info) {
+    std::vector<std::pair<storage::LabelId, storage::LabelIndexStats>> label_stats;
+    label_stats.reserve(index_info.size());
+    std::for_each(index_info.begin(), index_info.end(),
+                  [execution_db_accessor, view, &label_stats](const storage::LabelId &label_id) {
+                    auto vertices = execution_db_accessor->Vertices(view, label_id);
+                    uint64_t no_vertices{0};
+                    uint64_t total_degree{0};
+                    std::for_each(vertices.begin(), vertices.end(),
+                                  [&total_degree, &no_vertices, &view](const auto &vertex) {
+                                    no_vertices++;
+                                    total_degree += *vertex.OutDegree(view) + *vertex.InDegree(view);
+                                  });
+
+                    auto average_degree =
+                        no_vertices > 0 ? static_cast<double>(total_degree) / static_cast<double>(no_vertices) : 0;
```

Comment:
```suggestion
                        no_vertices > 0 ? total_degree / static_cast<double>(no_vertices) : 0;
```

Reply:
Needed to cast this because of `clang-tidy` (narrowing conversion from uint64_t to double)